presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/dereference/DereferencePushdown.java

```diff
@@ -26,51 +23,49 @@
 
 import java.util.Collection;
 import java.util.List;
-import java.util.Map;
 import java.util.Set;
-import java.util.function.Function;
 import java.util.stream.Collectors;
 
 import static com.google.common.base.Verify.verify;
-import static com.google.common.collect.ImmutableMap.toImmutableMap;
 import static com.google.common.collect.Iterables.getOnlyElement;
 import static io.prestosql.sql.planner.SymbolsExtractor.extractAll;
 
 /**
  * Provides helper methods to push down dereferences in the query plan.
  */
-class DereferencePushdown
+public class DereferencePushdown
```

Comment:
@martint I made this public since we also use one method from this class in PushLimitThroughProject and PushTopNThroughProject. This isn't ideal, should we pull out all classes from `io.prestosql.sql.planner.iterative.rule.dereference` to `io.prestosql.sql.planner.iterative.rule` package instead?

Reply:
Yeah, let’s move them up